docs/explainer.md

```diff
@@ -361,8 +334,36 @@ dictionary UDPMessage {
 
 See [`ReadableStream`](https://streams.spec.whatwg.org/#rs-intro) spec for more examples.
 
+##### Connected mode
+
+```javascript
+let udpSocket = new UDPSocket({ remoteAddress, remotePort });
+let { readable } = await udpSocket.opened;
+
+let reader = readable.getReader();
+
+let { value, done } = await reader.read();
+if (done) {
+  // stream is exhausted...
+  // happens either on socket errors or explicit socket.close();
```

Comment:
UDP sockets and TCP server sockets can never be exhausted and `socket.close()` cannot be called while the reader is locked, which effectively means that `read()` will never return with `done == true`. The only way these socket types can be closed while they are being read from is if there is a network error, which means `read()` throws an exception.

Only TCP sockets can be exhausted when the remote peer closes the connection.